modules/coreos-layering-configuring.adoc

```diff
@@ -0,0 +1,191 @@
+====
+
+* You must push the custom layered image to a repository that your cluster can access.
+
+.Procedure
+
+. Create a machine config file.
+
+.. Create a YAML file similar to the following:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker <1>
+  name: os-layer-hotfix
+spec:
+  osImageURL: quay.io/my-registry/custom-image@sha256:306b606615dcf8f0e5e7d87fee3 <2>
+----
+<1> Specifies the machine config pool to apply the custom layered image.
+<2> Specifies the path to the custom layered image.
+
+.. Create the `MachineConfig` object:
++
+[source,terminal]
+----
+$ oc create -f <file_name>.yaml
+----
++
+[IMPORTANT]
+====
+It is strongly recommended that you test your images outside of your production environment before rolling out to your cluster.
+====
+
+.Verification
+
+You can verify that the custom layered image is applied by performing any of the following checks:
+
+. Check that the worker machine config pool has rolled out with the new machine config:
+
+.. Check that the new machine config is created:
++
+[source,terminal]
+----
+$ oc get mc
+----
++
+.Sample output
+[source,terminal]
+----
+NAME                                               GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
+00-master                                          4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+00-worker                                          4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+01-master-container-runtime                        4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+01-master-kubelet                                  4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+01-worker-container-runtime                        4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+01-worker-kubelet                                  4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+99-master-generated-registries                     4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+99-master-ssh                                                                                 3.2.0             42m
+99-worker-generated-registries                     4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+99-worker-ssh                                                                                 3.2.0             42m
+os-layer-hotfix                                                                                                 5s <1>
+rendered-master-6faecdfa1b25c114a58cf178fbaa45e2   4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+rendered-worker-6b000dbc31aaee63c6a2d56d04cd4c1b   4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             37m
+rendered-worker-d41cef9c1658d2290bfe48480b890467   4c66f8fbbaa6de34f1252c2e1aebfaac21cfaa70   3.2.0             30s <2>
```

Comment:
These values are not coherent.

If the `os-layered-hotfix` MachineConfig generated the rendered MachineConfig `rendered-worker-d41cef9c1658d2290bfe48480b890467`, then `os-layered-hotfix` should be created before `rendered-worker-d41cef9c1658d2290bfe48480b890467`.

In this example, we see that `os-layer-hotfix ` was created 5 seconds ago, and `rendered-worker-d41cef9c1658d2290bfe48480b890467` was created 30 seconds ago. A solution could be to modify the "5s" in "os-layer-hotfix" and use "50s" instead.

`os-layer-hotfix` must be created before the resulting rendered MachineConfig.

Reply:
Good catch! Thank you.